### PR TITLE
Draft of form to create a game

### PR DESCRIPTION
**Pull request by saramissak**
This form allows the user to switch between stages to add the hints for the map. It also allows the user to add a description and title to the page. I don't have yet that the user can input the location onto the map.

### CODE REVIEW

File: src/main/webapp/js/createGame.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const activeStages = document.getElementsByClassName('activeStage')[0];
+  activeStages.classList.remove('activeStage');
+  const divToHide = activeStages.classList[0];
+  document.getElementById(divToHide + 'Hints').classList.add('hidden');
+  const stagesList = document.getElementById('stages');
+  const numStages = stagesList.getElementsByTagName('input').length;
+
+  const newStage = document.createElement('input');
+  newStage.value = 'Stage ' + (numStages + 1);
+  newStage.type= 'button';
```
Comment (ldcWV) [inline src/main/webapp/js/createGame.js]: Missing space

Comment (saramissak) [reply src/main/webapp/js/createGame.js]: Done, thanks.

File: target/portfolio-1/js/createGame.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const newStageHints = document.createElement('div');
+  newStageHints.id = newStage.classList[0] + 'Hints';
+  newStageHints.innerHTML = "<input type='text' placeholder='Starter (spawn location and first hint)' id='starter-hint' required>" +
+                            "<input type='text' placeholder='Hint 1' id='hint1'>";
+
+  document.getElementById('hints').appendChild(newStageHints);
+}
+
+function addNewHint() {
+  const activeStageNum = (document.getElementsByClassName('activeStage')[0]).classList[0] + 'Hints';
```
Comment (ldcWV) [inline target/portfolio-1/js/createGame.js:34]: Might want to write helper functions getActiveStageElement() and getStageNumber(stageElement) to make stuff like this clearer.

File: target/portfolio-1/js/createGame.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  activeStages.classList.remove('activeStage');
+  const divToHide = activeStages.classList[0];
+  document.getElementById(divToHide + 'Hints').classList.add('hidden');
+  const stagesList = document.getElementById('stages');
+  const numStages = stagesList.getElementsByTagName('input').length;
+
+  const newStage = document.createElement('input');
+  newStage.value = 'Stage ' + (numStages + 1);
+  newStage.type= 'button';
+  newStage.className = 'stage' + (numStages + 1) + ' activeStage';
```
Comment (ldcWV) [inline target/portfolio-1/js/createGame.js]: Use setActive() instead of this?

**Review by ldcWV — APPROVED** [review]

File: target/portfolio-1/js/createGame.js (hunk 2 above)
Comment (saramissak) [reply target/portfolio-1/js/createGame.js:34]: Okay, sounds good!

File: target/portfolio-1/js/createGame.js (hunk 3 above)
Comment (saramissak) [reply target/portfolio-1/js/createGame.js]: This is a good idea.

**Pull request merged**